ui/src/ReactCustomHooks/useWebSocket/useWebSocket.hook.js

```diff
@@ -0,0 +1,113 @@
+import { useState, useEffect } from 'react';
+import { NO_OP, EMPTY_OBJ } from 'Utils';
+
+const WEBSOCKET = {
+  CONNECTING: 0,
+  OPEN: 1,
+  CLOSING: 2,
+  CLOSED: 3,
+  INVALID: 'Invalid WebSocket function provided. See documentation',
+  SENT: true,
+  NOT_SENT: false,
+};
+
+const SOCKET_NOT_OPEN_HANDLER = () => {
+  // eslint-disable-next-line no-console
+  console.error('Socket is not open');
+  return WEBSOCKET.NOT_SENT;
+};
+
+const DEFAULT_STATE = {
+  send: SOCKET_NOT_OPEN_HANDLER,
+  currentState: WEBSOCKET.CLOSED,
+};
+
+const openWebSocket = (
+  getWebsocket,
+  { onOpen = NO_OP, onMessage = NO_OP, onError, onClose = NO_OP },
+  socketState,
+  updateSocketState
+) => {
+  let closeSocket = NO_OP;
+
+  if (
+    typeof getWebsocket === 'function' &&
+    socketState.currentState !== WebSocket.OPEN
+  ) {
+    const socket = getWebsocket();
+
+    closeSocket = () => {
+      socket.close(); //trigger the close of our socket
+      updateSocketState({ ...socketState, currentState: WEBSOCKET.CLOSING });
```

Comment:
The websocket has a `readyState` property - can we return that instead, incase the constants change at any point.

Reply:
I will reply here (and refer to this comment in the others like this). I think this is more readable/obvious to what the state is changing to, vs say `socket.readyStatus`. It would be the same value in both cases - I just feel this is more explicit.